packages/astro/src/vite-plugin-astro/index.ts

```diff
@@ -215,14 +213,6 @@ export default function astro({ config, logging }: AstroPluginOptions): vite.Plu
 			}
 
 			const filename = normalizeFilename(parsedId.filename);
-			let isPage = false;
-			try {
-				const fileUrl = new URL(`file://${filename}`);
-				isPage = fileUrl.pathname.startsWith(resolvePages(config).pathname);
-			} catch {}
-			if (isPage && config._ctx.scripts.some((s) => s.stage === 'page')) {
-				source += `\n<script src="${PAGE_SCRIPT_ID}" />`;
```

Comment:
how is PAGE_SCRIPT_ID being added now? I see it being added to dev in render, but that wouldn't cause it to be added in the build.

Reply:
Great catch, working on this now!

Reply:
@matthewp fixed in https://github.com/withastro/astro/pull/4049/commits/99f20d1d837df9b1cf2b9aa7aa541d6609b0a6cf